### ISSUE TITLE
Question about constraints

### ISSUE THREAD

**Issue by rodrigomelo9** [SymbiFlow/symbiflow-examples#126]
Hi. It seems that both, `pcf` and `xdc` are supported.

I was comparing the three xc7 examples. Taking for example `arty`, I found:
* `counter_test/arty.xdc`
* `linux_litex_demo`:
    * `arty.pcf`
    * `arty.xcf`
* `picosoc_demo/arty.pcf`

Comparing the `Makefiles`:
* For `symbiflow_synth`, `counter_test` and `linux_litex_demo` specify `-x ${XDC}` (don't used by `picosoc_demo`)
* For `symbiflow_place`, `linux_litex_demo` and `picosoc_demo` specify `-p ${PCF}` (don't used by `counter_test`)

Is the same to use `-x ${XDC}` for `symbiflow_synth` and `-p ${PCF}` for `symbiflow_place`? Do I need to choose only one option? Has it sense to support the two alternatives? Recommendation? In my opinion, in place has more sense (but is using pcf instead of xdc). Is it taken into account for synthesis?).

Clarification will be welcome.

Regards,
Rodrigo

**Comment by litghost** [comment]
We are moving to XDC, but some of the old demos were written when only PCF support existed.

**Comment by rodrigomelo9** [comment]
Ok, great. What about to specify in `symbiflow_synth` instead of `symbiflow_place`? Is it the final destination? (`symbiflow_synth`?) There, we can specify the timing and place constraints?

**Comment by mithro** [comment]
I think that is just a mistake?

**Comment by mithro** [comment]
@acomodi / @kgugala -- Thoughts?

**Comment by acomodi** [comment]
@rodrigomelo9 It actually is expected it to be like that.

- PCF: this type of constraints is only concerning placement of IOs, therefore it is required during `symbiflow_place` as the placement constraints are read at that step only (and a constraints file tailored for VPR is generated as output)
- XDC: this file can contain different kind of constraints and not only placement ones (e.g. timing, etc). We have a yosys plugin capable of parsing the XDC constraints and save the placement locations of the IOs directly in the eblif, thus not requiring any additional arguments in the `symbiflow_place` step.

Additionally, XDC and PCF should not be used at the same time and it is better to use XDC, given that they can contain also timing and other constraints.

**Comment by mithro** [comment]
@acomodi -- This seems incorrect. The XDC should contain the IO placements (as well as other constraints)? Is the problem that we are missing a tool which generates the IO constraints file for VPR from an XDC input?

**Comment by acomodi** [comment]
@mithro This is what is happening, the XDC contains all kinds of constraints. The yosys plugin can read them and save them in the eblif, for the io place constraints script to generate the VPR constraint files

**Comment by mithro** [comment]
@acomodi -- So the PCF file is being generated by the output of Yosys which is then converted into the correct format needed by VPR?

**Comment by acomodi** [comment]
@mithro No. There are two independent flows concerning IO placement constraints, depending on what the user chooses:
XDC flow: yosys parses the file and saves the constraints in the eblif. The placement constraints script will read from the eblif and generate the VPR constraints. No PCF is produced
PCF flow: yosys does nothing. The placement constraints script reads the PCF and generates the VPR constraints. (The eblif is read also here, but only to get the top level IO nets)

**Comment by mithro** [comment]
I think we might have discussed this previously. We should hand in the PCF at the same place we hand in the XDC and then the rest of the flow should be the same.

See https://github.com/SymbiFlow/yosys-symbiflow-plugins/issues/30 and maybe https://docs.google.com/drawings/d/1r2LXypJF5AD40LfHegml3_fIvPT2jZ3n2OZYW9-9dLU/edit ?

**Comment by rodrigomelo9** [comment]
Thanks. My question was answered, so this issue can be closed if you want (I mean, it seems that @mithro detected another problem, but that is not what this issue was about).

Regards
